daemon/entrypoint.sh

```diff
@@ -73,6 +74,10 @@ if [[ "$CEPH_DAEMON" = "MON" ]]; then
     exit 1
   fi
 
+  if [ ${MON_IP_AUTO_DETECT} -eq 1 ]; then
```

Comment:
I'm going to be a pain about this:  I propose that `MON_IP_AUTO_DETECT` can be set to various options:
*  0 = disabled
*  1 = enabled, automatic (IPv6, falling back to IPv4)
*  4 = enabled, IPv4
*  6 = enabled, IPv6